internal/controller/drplacementcontrol.go

```diff
@@ -1186,22 +1194,48 @@ func (d *DRPCInstance) readyToSwitchOver(homeCluster string, preferredCluster st
 		}
 	}
 	// Allow switch over when PV data is ready and the cluster data is protected
-	return d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady) &&
-		d.isVRGConditionMet(homeCluster, VRGConditionTypeClusterDataProtected)
+	dataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady)
+	if err != nil || !dataReady {
+		return false
+	}
+
+	clusterDataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeClusterDataProtected)
+
+	return err == nil && clusterDataReady
```

Comment:
How err can be non nil here? we return in line 1199 in this case.

Reply:
It is another error - in line 1199 we checked VRGConditionTypeDataReady condition and here is VRGConditionTypeClusterDataProtected condition

Reply:
Right but we drop this error silently again.

Reply:
@nirs both errors are logged now